src/devices/Nmea0183/Sentences/HeadingAndDeclination.cs

```diff
@@ -0,0 +1,191 @@
+            {
+                HeadingTrue = Angle.FromDegrees(heading.Value);
+                // This one needs to be there, the others are optional
+                Valid = true;
+            }
+
+            if (deviation.HasValue)
+            {
+                if (deviationDirection == "E")
+                {
+                    Deviation = Angle.FromDegrees(deviation.Value);
+                }
+                else
+                {
+                    Deviation = Angle.FromDegrees(deviation.Value * -1);
+                }
+            }
+
+            if (magVar.HasValue)
+            {
+                if (magVarDirection == "E")
+                {
+                    Declination = Angle.FromDegrees(magVar.Value);
+                }
+                else
+                {
+                    Declination = Angle.FromDegrees(magVar.Value * -1);
+                }
+            }
+        }
+
+        /// <summary>
+        /// This is true for this message type
+        /// </summary>
+        public override bool ReplacesOlderInstance => true;
+
+        /// <summary>
+        /// Angle of the wind
+        /// </summary>
+        public Angle HeadingTrue
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Magnetic heading (derived from true heading and declination)
+        /// </summary>
+        public Angle? HeadingMagnetic
+        {
+            get
+            {
+                if (Declination.HasValue)
+                {
+                    return (HeadingTrue - Declination.Value).Normalize(true);
+                }
+
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Deviation at current location. Usually unknown (empty). Not sure what this field means. The deviation is a property
+        /// of the actual compass and the current orientation, not the location.
+        /// </summary>
+        public Angle? Deviation
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Magnetic declination (sometimes also called variation) at current location. Usually derived from the NOAA magnetic field model by one
+        /// of the attached devices.
+        /// </summary>
+        public Angle? Declination
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Presents this message as output
+        /// </summary>
+        public override string ToNmeaParameterList()
+        {
+            if (Valid)
+            {
+                StringBuilder b = new StringBuilder();
+                b.AppendFormat(CultureInfo.InvariantCulture, "{0:F1},", HeadingTrue.Normalize(true).Degrees);
+                if (Deviation.HasValue)
+                {
+                    b.AppendFormat(CultureInfo.InvariantCulture, "{0:F1},{1},", Math.Abs(Deviation.Value.Degrees),
+                        Deviation.Value.Degrees >= 0 ? "E" : "W");
```

Comment:
You are using quite often "E" and "W" and "N" and "S" all up. Does it makes sense to create a global const or char enum? You even have the `CardinalDirection` as internal byte.